pytorch_lightning/loops/epoch/training_epoch_loop.py

```diff
@@ -76,7 +76,14 @@ def done(self) -> bool:
         signals to stop (e.g. by early stopping).
         """
         max_steps_reached = self.max_steps is not None and self.global_step >= self.max_steps
-        return max_steps_reached or self.trainer.should_stop or self._num_training_batches_reached()
+        # done should be True only when `batch_loop` and `val_loop` are done.
+        should_check_val = self.restarting and self._should_check_val_fx(self.batch_idx, self.is_last_batch)
+        return (
+            max_steps_reached
+            or self.trainer.should_stop
+            or self._num_training_batches_reached()
+            and not should_check_val
+        )
```

Comment:
may consider use var assignment, not this long return which may be handier in case of debugging and placing breakpoint...

Reply:
The values can be seen by placing the checkpoint in the return line